{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv('GSE148375_processed_data.txt', delimiter='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID_REF</th>\n",
       "      <th>1900017</th>\n",
       "      <th>1900029</th>\n",
       "      <th>1900030</th>\n",
       "      <th>1900031</th>\n",
       "      <th>1900040</th>\n",
       "      <th>1900043</th>\n",
       "      <th>1900046</th>\n",
       "      <th>1900048</th>\n",
       "      <th>1900049</th>\n",
       "      <th>...</th>\n",
       "      <th>215606</th>\n",
       "      <th>215753</th>\n",
       "      <th>215762</th>\n",
       "      <th>215763</th>\n",
       "      <th>215764</th>\n",
       "      <th>215769</th>\n",
       "      <th>215771</th>\n",
       "      <th>215772</th>\n",
       "      <th>215773</th>\n",
       "      <th>215827</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>exm2268640-0_B_F_1984844585</td>\n",
       "      <td>AG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>...</td>\n",
       "      <td>GG</td>\n",
       "      <td>NC</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>exm41-0_B_F_1921435147</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>...</td>\n",
       "      <td>GG</td>\n",
       "      <td>AA</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>exm1916089-0_B_R_1927689775</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>...</td>\n",
       "      <td>GG</td>\n",
       "      <td>NC</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>exm44-0_B_R_1921538602</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>...</td>\n",
       "      <td>GG</td>\n",
       "      <td>AG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>exm46-0_T_F_1921333919</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>...</td>\n",
       "      <td>GG</td>\n",
       "      <td>NC</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 3400 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                        ID_REF 1900017 1900029 1900030 1900031 1900040  \\\n",
       "0  exm2268640-0_B_F_1984844585      AG      GG      GG      GG      GG   \n",
       "1       exm41-0_B_F_1921435147      GG      GG      GG      GG      GG   \n",
       "2  exm1916089-0_B_R_1927689775      GG      GG      GG      GG      GG   \n",
       "3       exm44-0_B_R_1921538602      GG      GG      GG      GG      GG   \n",
       "4       exm46-0_T_F_1921333919      GG      GG      GG      GG      GG   \n",
       "\n",
       "  1900043 1900046 1900048 1900049  ... 215606 215753 215762 215763 215764  \\\n",
       "0      GG      GG      GG      GG  ...     GG     NC     GG     GG     GG   \n",
       "1      GG      GG      GG      GG  ...     GG     AA     GG     GG     GG   \n",
       "2      GG      GG      GG      GG  ...     GG     NC     GG     GG     GG   \n",
       "3      GG      GG      GG      GG  ...     GG     AG     GG     GG     GG   \n",
       "4      GG      GG      GG      GG  ...     GG     NC     GG     GG     GG   \n",
       "\n",
       "  215769 215771 215772 215773 215827  \n",
       "0     GG     GG     GG     GG     GG  \n",
       "1     GG     GG     GG     GG     GG  \n",
       "2     GG     GG     GG     GG     GG  \n",
       "3     GG     GG     GG     GG     GG  \n",
       "4     GG     GG     GG     GG     GG  \n",
       "\n",
       "[5 rows x 3400 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(242901, 3400)"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "import csv\n",
    "\n",
    "# Chemin vers le fichier CSV d'origine\n",
    "chemin_fichier_origine = \"GPL18545_humanexome-12v1-1_a.csv\"\n",
    "\n",
    "# Chemin vers le nouveau fichier CSV sans les 8 premières lignes\n",
    "chemin_nouveau_fichier = \"newGPL18545_humanexome-12v1-1_a.csv\"\n",
    "\n",
    "# Ouvre le fichier CSV d'origine et lit son contenu\n",
    "with open(chemin_fichier_origine, newline='') as csvfile_origine:\n",
    "    contenu_csv = csv.reader(csvfile_origine)\n",
    "\n",
    "    # Lit et ignore les 8 premières lignes du fichier CSV\n",
    "    for i in range(7):\n",
    "        next(contenu_csv)\n",
    "\n",
    "    # Crée un nouveau fichier CSV sans les 8 premières lignes\n",
    "    with open(chemin_nouveau_fichier, mode='w', newline='') as csvfile_nouveau:\n",
    "        csv_writer = csv.writer(csvfile_nouveau)\n",
    "\n",
    "        # Copie le contenu du fichier CSV d'origine (sans les 8 premières lignes) vers le nouveau fichier CSV\n",
    "        for row in contenu_csv:\n",
    "            csv_writer.writerow(row)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_csv= pd.read_csv('newGPL18545_humanexome-12v1-1_a.csv', delimiter=\",\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(242925, 21)"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_csv.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>IlmnID</th>\n",
       "      <th>Name</th>\n",
       "      <th>IlmnStrand</th>\n",
       "      <th>SNP</th>\n",
       "      <th>AddressA_ID</th>\n",
       "      <th>AlleleA_ProbeSeq</th>\n",
       "      <th>AddressB_ID</th>\n",
       "      <th>AlleleB_ProbeSeq</th>\n",
       "      <th>GenomeBuild</th>\n",
       "      <th>Chr</th>\n",
       "      <th>...</th>\n",
       "      <th>Ploidy</th>\n",
       "      <th>Species</th>\n",
       "      <th>Source</th>\n",
       "      <th>SourceVersion</th>\n",
       "      <th>SourceStrand</th>\n",
       "      <th>SourceSeq</th>\n",
       "      <th>TopGenomicSeq</th>\n",
       "      <th>BeadSetID</th>\n",
       "      <th>Exp_Clusters</th>\n",
       "      <th>RefStrand</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>exm-IND1-200449980-0_M_R_1990486447</td>\n",
       "      <td>exm-IND1-200449980</td>\n",
       "      <td>MINUS</td>\n",
       "      <td>[D/I]</td>\n",
       "      <td>69606251.0</td>\n",
       "      <td>CTGTGGAACTTTCCGGAGCTCGGCACCCTTCCCCCAAGCCTACCTG...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>37.1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>diploid</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>1000genomes</td>\n",
       "      <td>0.0</td>\n",
       "      <td>PLUS</td>\n",
       "      <td>GAGGGCCGCTCAGCGAGGGCGGGACAGAACCTCTCCCGGGCTGGGA...</td>\n",
       "      <td>GAGGGCCGCTCAGCGAGGGCGGGACAGAACCTCTCCCGGGCTGGGA...</td>\n",
       "      <td>662.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>exm-IND1-201453487-0_M_R_1990486448</td>\n",
       "      <td>exm-IND1-201453487</td>\n",
       "      <td>MINUS</td>\n",
       "      <td>[D/I]</td>\n",
       "      <td>3775555.0</td>\n",
       "      <td>TCCTGCAACCAGGGCCGATACCCCCTCATCCAGACGCTACGGCAGG...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>37.1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>diploid</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>1000genomes</td>\n",
       "      <td>0.0</td>\n",
       "      <td>PLUS</td>\n",
       "      <td>CATGGAGCTGTGTTTGTACCCCACCTCCAAATTCCACCACTGGCCC...</td>\n",
       "      <td>CATGGAGCTGTGTTTGTACCCCACCTCCAAATTCCACCACTGGCCC...</td>\n",
       "      <td>662.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>exm-IND1-85310248-0_P_F_1990486624</td>\n",
       "      <td>exm-IND1-85310248</td>\n",
       "      <td>PLUS</td>\n",
       "      <td>[I/D]</td>\n",
       "      <td>56795122.0</td>\n",
       "      <td>TCAGAACCATGGCTCCAAAACAAAAGAAAAAGACATCACGTGGCAA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>37.1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>diploid</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>1000genomes</td>\n",
       "      <td>0.0</td>\n",
       "      <td>PLUS</td>\n",
       "      <td>CTTTATGCAGCCCAAGTCAGAACCATGGCTCCAAAACAAAAGAAAA...</td>\n",
       "      <td>CTTTATGCAGCCCAAGTCAGAACCATGGCTCCAAAACAAAAGAAAA...</td>\n",
       "      <td>688.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>exm-IND10-102817747-0_P_F_1990486569</td>\n",
       "      <td>exm-IND10-102817747</td>\n",
       "      <td>PLUS</td>\n",
       "      <td>[I/D]</td>\n",
       "      <td>30777118.0</td>\n",
       "      <td>CTCTGCAAGTTTTGTGCCTTAGCGGAAAGAGGCGACATCTCTGGCA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>37.1</td>\n",
       "      <td>10</td>\n",
       "      <td>...</td>\n",
       "      <td>diploid</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>1000genomes</td>\n",
       "      <td>0.0</td>\n",
       "      <td>PLUS</td>\n",
       "      <td>GGGAACACCTGGCTCTGCAAGTTTTGTGCCTTAGCGGAAAGAGGCG...</td>\n",
       "      <td>GGGAACACCTGGCTCTGCAAGTTTTGTGCCTTAGCGGAAAGAGGCG...</td>\n",
       "      <td>662.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>exm-IND10-18329639-0_M_R_1990486442</td>\n",
       "      <td>exm-IND10-18329639</td>\n",
       "      <td>MINUS</td>\n",
       "      <td>[D/I]</td>\n",
       "      <td>87684402.0</td>\n",
       "      <td>AAGGCTGCTCCTATGGCTAGGCCATCTGCAAAATTATGCAGGCTGT...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>37.1</td>\n",
       "      <td>10</td>\n",
       "      <td>...</td>\n",
       "      <td>diploid</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>1000genomes</td>\n",
       "      <td>0.0</td>\n",
       "      <td>PLUS</td>\n",
       "      <td>CTGTGTTTATTTGGTTCCTTAGGTAAAGCCATTAGCTTGTTAGCAA...</td>\n",
       "      <td>CTGTGTTTATTTGGTTCCTTAGGTAAAGCCATTAGCTTGTTAGCAA...</td>\n",
       "      <td>662.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                 IlmnID                 Name IlmnStrand  \\\n",
       "0   exm-IND1-200449980-0_M_R_1990486447   exm-IND1-200449980      MINUS   \n",
       "1   exm-IND1-201453487-0_M_R_1990486448   exm-IND1-201453487      MINUS   \n",
       "2    exm-IND1-85310248-0_P_F_1990486624    exm-IND1-85310248       PLUS   \n",
       "3  exm-IND10-102817747-0_P_F_1990486569  exm-IND10-102817747       PLUS   \n",
       "4   exm-IND10-18329639-0_M_R_1990486442   exm-IND10-18329639      MINUS   \n",
       "\n",
       "     SNP  AddressA_ID                                   AlleleA_ProbeSeq  \\\n",
       "0  [D/I]   69606251.0  CTGTGGAACTTTCCGGAGCTCGGCACCCTTCCCCCAAGCCTACCTG...   \n",
       "1  [D/I]    3775555.0  TCCTGCAACCAGGGCCGATACCCCCTCATCCAGACGCTACGGCAGG...   \n",
       "2  [I/D]   56795122.0  TCAGAACCATGGCTCCAAAACAAAAGAAAAAGACATCACGTGGCAA...   \n",
       "3  [I/D]   30777118.0  CTCTGCAAGTTTTGTGCCTTAGCGGAAAGAGGCGACATCTCTGGCA...   \n",
       "4  [D/I]   87684402.0  AAGGCTGCTCCTATGGCTAGGCCATCTGCAAAATTATGCAGGCTGT...   \n",
       "\n",
       "   AddressB_ID AlleleB_ProbeSeq  GenomeBuild Chr  ...   Ploidy       Species  \\\n",
       "0          NaN              NaN         37.1   1  ...  diploid  Homo sapiens   \n",
       "1          NaN              NaN         37.1   1  ...  diploid  Homo sapiens   \n",
       "2          NaN              NaN         37.1   1  ...  diploid  Homo sapiens   \n",
       "3          NaN              NaN         37.1  10  ...  diploid  Homo sapiens   \n",
       "4          NaN              NaN         37.1  10  ...  diploid  Homo sapiens   \n",
       "\n",
       "        Source SourceVersion  SourceStrand  \\\n",
       "0  1000genomes           0.0          PLUS   \n",
       "1  1000genomes           0.0          PLUS   \n",
       "2  1000genomes           0.0          PLUS   \n",
       "3  1000genomes           0.0          PLUS   \n",
       "4  1000genomes           0.0          PLUS   \n",
       "\n",
       "                                           SourceSeq  \\\n",
       "0  GAGGGCCGCTCAGCGAGGGCGGGACAGAACCTCTCCCGGGCTGGGA...   \n",
       "1  CATGGAGCTGTGTTTGTACCCCACCTCCAAATTCCACCACTGGCCC...   \n",
       "2  CTTTATGCAGCCCAAGTCAGAACCATGGCTCCAAAACAAAAGAAAA...   \n",
       "3  GGGAACACCTGGCTCTGCAAGTTTTGTGCCTTAGCGGAAAGAGGCG...   \n",
       "4  CTGTGTTTATTTGGTTCCTTAGGTAAAGCCATTAGCTTGTTAGCAA...   \n",
       "\n",
       "                                       TopGenomicSeq BeadSetID  Exp_Clusters  \\\n",
       "0  GAGGGCCGCTCAGCGAGGGCGGGACAGAACCTCTCCCGGGCTGGGA...     662.0           3.0   \n",
       "1  CATGGAGCTGTGTTTGTACCCCACCTCCAAATTCCACCACTGGCCC...     662.0           3.0   \n",
       "2  CTTTATGCAGCCCAAGTCAGAACCATGGCTCCAAAACAAAAGAAAA...     688.0           3.0   \n",
       "3  GGGAACACCTGGCTCTGCAAGTTTTGTGCCTTAGCGGAAAGAGGCG...     662.0           3.0   \n",
       "4  CTGTGTTTATTTGGTTCCTTAGGTAAAGCCATTAGCTTGTTAGCAA...     662.0           3.0   \n",
       "\n",
       "   RefStrand  \n",
       "0          -  \n",
       "1          -  \n",
       "2          +  \n",
       "3          +  \n",
       "4          -  \n",
       "\n",
       "[5 rows x 21 columns]"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_csv.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "import csv\n",
    "\n",
    "# Ouverture du fichier CSV en mode lecture\n",
    "with open('GPL18545_humanexome-12v1-1_a.csv', 'r') as csvfile:\n",
    "    # Lecture du contenu du fichier CSV\n",
    "    csvreader = csv.reader(csvfile)\n",
    "\n",
    "    # Suppression des 7 premières lignes\n",
    "    for i in range(7):\n",
    "        next(csvreader)\n",
    "\n",
    "    # Ouverture du fichier CSV en mode écriture\n",
    "    with open('nouveau_fichier.csv', 'w', newline='') as newcsvfile:\n",
    "        # Écriture du contenu du fichier CSV sans les 7 premières lignes\n",
    "        csvwriter = csv.writer(newcsvfile)\n",
    "        for row in csvreader:\n",
    "            csvwriter.writerow(row)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/local/anaconda3/lib/python3.6/site-packages/IPython/core/interactiveshell.py:3057: DtypeWarning: Columns (9) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    }
   ],
   "source": [
    "input_file = \"nouveau_fichier.csv\"\n",
    "genotypes = pd.read_csv(input_file, sep=\",\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>IlmnID</th>\n",
       "      <th>Name</th>\n",
       "      <th>IlmnStrand</th>\n",
       "      <th>SNP</th>\n",
       "      <th>AddressA_ID</th>\n",
       "      <th>AlleleA_ProbeSeq</th>\n",
       "      <th>AddressB_ID</th>\n",
       "      <th>AlleleB_ProbeSeq</th>\n",
       "      <th>GenomeBuild</th>\n",
       "      <th>Chr</th>\n",
       "      <th>...</th>\n",
       "      <th>Ploidy</th>\n",
       "      <th>Species</th>\n",
       "      <th>Source</th>\n",
       "      <th>SourceVersion</th>\n",
       "      <th>SourceStrand</th>\n",
       "      <th>SourceSeq</th>\n",
       "      <th>TopGenomicSeq</th>\n",
       "      <th>BeadSetID</th>\n",
       "      <th>Exp_Clusters</th>\n",
       "      <th>RefStrand</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>exm-IND1-200449980-0_M_R_1990486447</td>\n",
       "      <td>exm-IND1-200449980</td>\n",
       "      <td>MINUS</td>\n",
       "      <td>[D/I]</td>\n",
       "      <td>69606251.0</td>\n",
       "      <td>CTGTGGAACTTTCCGGAGCTCGGCACCCTTCCCCCAAGCCTACCTG...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>37.1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>diploid</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>1000genomes</td>\n",
       "      <td>0.0</td>\n",
       "      <td>PLUS</td>\n",
       "      <td>GAGGGCCGCTCAGCGAGGGCGGGACAGAACCTCTCCCGGGCTGGGA...</td>\n",
       "      <td>GAGGGCCGCTCAGCGAGGGCGGGACAGAACCTCTCCCGGGCTGGGA...</td>\n",
       "      <td>662.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>exm-IND1-201453487-0_M_R_1990486448</td>\n",
       "      <td>exm-IND1-201453487</td>\n",
       "      <td>MINUS</td>\n",
       "      <td>[D/I]</td>\n",
       "      <td>3775555.0</td>\n",
       "      <td>TCCTGCAACCAGGGCCGATACCCCCTCATCCAGACGCTACGGCAGG...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>37.1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>diploid</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>1000genomes</td>\n",
       "      <td>0.0</td>\n",
       "      <td>PLUS</td>\n",
       "      <td>CATGGAGCTGTGTTTGTACCCCACCTCCAAATTCCACCACTGGCCC...</td>\n",
       "      <td>CATGGAGCTGTGTTTGTACCCCACCTCCAAATTCCACCACTGGCCC...</td>\n",
       "      <td>662.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>exm-IND1-85310248-0_P_F_1990486624</td>\n",
       "      <td>exm-IND1-85310248</td>\n",
       "      <td>PLUS</td>\n",
       "      <td>[I/D]</td>\n",
       "      <td>56795122.0</td>\n",
       "      <td>TCAGAACCATGGCTCCAAAACAAAAGAAAAAGACATCACGTGGCAA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>37.1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>diploid</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>1000genomes</td>\n",
       "      <td>0.0</td>\n",
       "      <td>PLUS</td>\n",
       "      <td>CTTTATGCAGCCCAAGTCAGAACCATGGCTCCAAAACAAAAGAAAA...</td>\n",
       "      <td>CTTTATGCAGCCCAAGTCAGAACCATGGCTCCAAAACAAAAGAAAA...</td>\n",
       "      <td>688.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>exm-IND10-102817747-0_P_F_1990486569</td>\n",
       "      <td>exm-IND10-102817747</td>\n",
       "      <td>PLUS</td>\n",
       "      <td>[I/D]</td>\n",
       "      <td>30777118.0</td>\n",
       "      <td>CTCTGCAAGTTTTGTGCCTTAGCGGAAAGAGGCGACATCTCTGGCA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>37.1</td>\n",
       "      <td>10</td>\n",
       "      <td>...</td>\n",
       "      <td>diploid</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>1000genomes</td>\n",
       "      <td>0.0</td>\n",
       "      <td>PLUS</td>\n",
       "      <td>GGGAACACCTGGCTCTGCAAGTTTTGTGCCTTAGCGGAAAGAGGCG...</td>\n",
       "      <td>GGGAACACCTGGCTCTGCAAGTTTTGTGCCTTAGCGGAAAGAGGCG...</td>\n",
       "      <td>662.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>exm-IND10-18329639-0_M_R_1990486442</td>\n",
       "      <td>exm-IND10-18329639</td>\n",
       "      <td>MINUS</td>\n",
       "      <td>[D/I]</td>\n",
       "      <td>87684402.0</td>\n",
       "      <td>AAGGCTGCTCCTATGGCTAGGCCATCTGCAAAATTATGCAGGCTGT...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>37.1</td>\n",
       "      <td>10</td>\n",
       "      <td>...</td>\n",
       "      <td>diploid</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>1000genomes</td>\n",
       "      <td>0.0</td>\n",
       "      <td>PLUS</td>\n",
       "      <td>CTGTGTTTATTTGGTTCCTTAGGTAAAGCCATTAGCTTGTTAGCAA...</td>\n",
       "      <td>CTGTGTTTATTTGGTTCCTTAGGTAAAGCCATTAGCTTGTTAGCAA...</td>\n",
       "      <td>662.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                 IlmnID                 Name IlmnStrand  \\\n",
       "0   exm-IND1-200449980-0_M_R_1990486447   exm-IND1-200449980      MINUS   \n",
       "1   exm-IND1-201453487-0_M_R_1990486448   exm-IND1-201453487      MINUS   \n",
       "2    exm-IND1-85310248-0_P_F_1990486624    exm-IND1-85310248       PLUS   \n",
       "3  exm-IND10-102817747-0_P_F_1990486569  exm-IND10-102817747       PLUS   \n",
       "4   exm-IND10-18329639-0_M_R_1990486442   exm-IND10-18329639      MINUS   \n",
       "\n",
       "     SNP  AddressA_ID                                   AlleleA_ProbeSeq  \\\n",
       "0  [D/I]   69606251.0  CTGTGGAACTTTCCGGAGCTCGGCACCCTTCCCCCAAGCCTACCTG...   \n",
       "1  [D/I]    3775555.0  TCCTGCAACCAGGGCCGATACCCCCTCATCCAGACGCTACGGCAGG...   \n",
       "2  [I/D]   56795122.0  TCAGAACCATGGCTCCAAAACAAAAGAAAAAGACATCACGTGGCAA...   \n",
       "3  [I/D]   30777118.0  CTCTGCAAGTTTTGTGCCTTAGCGGAAAGAGGCGACATCTCTGGCA...   \n",
       "4  [D/I]   87684402.0  AAGGCTGCTCCTATGGCTAGGCCATCTGCAAAATTATGCAGGCTGT...   \n",
       "\n",
       "   AddressB_ID AlleleB_ProbeSeq  GenomeBuild Chr  ...   Ploidy       Species  \\\n",
       "0          NaN              NaN         37.1   1  ...  diploid  Homo sapiens   \n",
       "1          NaN              NaN         37.1   1  ...  diploid  Homo sapiens   \n",
       "2          NaN              NaN         37.1   1  ...  diploid  Homo sapiens   \n",
       "3          NaN              NaN         37.1  10  ...  diploid  Homo sapiens   \n",
       "4          NaN              NaN         37.1  10  ...  diploid  Homo sapiens   \n",
       "\n",
       "        Source SourceVersion  SourceStrand  \\\n",
       "0  1000genomes           0.0          PLUS   \n",
       "1  1000genomes           0.0          PLUS   \n",
       "2  1000genomes           0.0          PLUS   \n",
       "3  1000genomes           0.0          PLUS   \n",
       "4  1000genomes           0.0          PLUS   \n",
       "\n",
       "                                           SourceSeq  \\\n",
       "0  GAGGGCCGCTCAGCGAGGGCGGGACAGAACCTCTCCCGGGCTGGGA...   \n",
       "1  CATGGAGCTGTGTTTGTACCCCACCTCCAAATTCCACCACTGGCCC...   \n",
       "2  CTTTATGCAGCCCAAGTCAGAACCATGGCTCCAAAACAAAAGAAAA...   \n",
       "3  GGGAACACCTGGCTCTGCAAGTTTTGTGCCTTAGCGGAAAGAGGCG...   \n",
       "4  CTGTGTTTATTTGGTTCCTTAGGTAAAGCCATTAGCTTGTTAGCAA...   \n",
       "\n",
       "                                       TopGenomicSeq BeadSetID  Exp_Clusters  \\\n",
       "0  GAGGGCCGCTCAGCGAGGGCGGGACAGAACCTCTCCCGGGCTGGGA...     662.0           3.0   \n",
       "1  CATGGAGCTGTGTTTGTACCCCACCTCCAAATTCCACCACTGGCCC...     662.0           3.0   \n",
       "2  CTTTATGCAGCCCAAGTCAGAACCATGGCTCCAAAACAAAAGAAAA...     688.0           3.0   \n",
       "3  GGGAACACCTGGCTCTGCAAGTTTTGTGCCTTAGCGGAAAGAGGCG...     662.0           3.0   \n",
       "4  CTGTGTTTATTTGGTTCCTTAGGTAAAGCCATTAGCTTGTTAGCAA...     662.0           3.0   \n",
       "\n",
       "   RefStrand  \n",
       "0          -  \n",
       "1          -  \n",
       "2          +  \n",
       "3          +  \n",
       "4          -  \n",
       "\n",
       "[5 rows x 21 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genotypes.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Chemin vers le fichier CSV contenant les données de génotypage Illumina\n",
    "input_file = \"nouveau_fichier.csv\"\n",
    "\n",
    "# Lecture des données de génotypage à partir du fichier CSV\n",
    "genotypes = pd.read_csv(input_file, sep=\",\")\n",
    "\n",
    "# Conversion des génotypes au format VCF\n",
    "vcf = pd.DataFrame(columns=[\"CHROM\", \"POS\", \"ID\", \"REF\", \"ALT\", \"QUAL\", \"FILTER\", \"INFO\", \"FORMAT\"])\n",
    "\n",
    "vcf[\"CHROM\"] = genotypes[\"Chr\"]\n",
    "vcf[\"POS\"] = genotypes[\"MapInfo\"]\n",
    "vcf[\"ID\"] = genotypes[\"IlmnID\"]\n",
    "vcf[\"REF\"] = genotypes[\"AlleleA_ProbeSeq\"]\n",
    "vcf[\"ALT\"] = genotypes[\"AlleleB_ProbeSeq\"]\n",
    "vcf[\"QUAL\"] = \".\"\n",
    "vcf[\"FILTER\"] = \".\"\n",
    "vcf[\"INFO\"] = \".\"\n",
    "vcf[\"FORMAT\"] = \"GT\"\n",
    "\n",
    "for i in range(0, len(genotypes.index)):\n",
    "    # Conversion des génotypes en format VCF\n",
    "    genotype = genotypes.iloc[i, 6:]\n",
    "    vcf.loc[i, \"SAMPLE\"] = \"Sample_\" + str(i + 1)\n",
    "    vcf.loc[i, \"SAMPLE\"] = \":\".join([str(x) for x in genotype])\n",
    "\n",
    "# Écriture des génotypes convertis en format VCF dans un fichier\n",
    "vcf.to_csv(\"genotypes.vcf\", sep=\"\\t\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Chemin vers le fichier CSV\n",
    "filename = \"newGPL18545_humanexome-12v1-1_a.csv\"\n",
    "\n",
    "# Charger le fichier CSV dans un dataframe pandas\n",
    "df = pd.read_csv(filename)\n",
    "\n",
    "# Supprimer les 23 dernières lignes\n",
    "df = df.iloc[:-1]\n",
    "\n",
    "# Écrire les données modifiées dans le fichier CSV\n",
    "df.to_csv(\"newGPL18545_humanexome-12v1-1_a.csv\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Une fonction qui supprime les crochéts\n",
    "def remove_ends(input_string):\n",
    "    if len(input_string) < 3:\n",
    "        return \"\"\n",
    "    else:\n",
    "        return input_string[1:-1]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "def remove_ends_column(df, column_name):\n",
    "    # Créer une nouvelle colonne avec les valeurs modifiées\n",
    "    new_column = []\n",
    "    for i in df.index:\n",
    "        value = df.loc[i, column_name]\n",
    "        if isinstance(value, str):\n",
    "            new_value = remove_ends(value)\n",
    "        else:\n",
    "            new_value = \"\"\n",
    "        new_column.append(new_value)\n",
    "    # Remplacer la colonne existante par la nouvelle colonne\n",
    "    df[column_name] = new_column\n",
    "    return df\n",
    "\n",
    "\n",
    "remove_ends_column(genotypes, \"SNP\").head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Une fonction permettant d'extraire les SNPs des individus connaissant leurs identifiant de référence\n",
    "def extract_row_by_id(df, id, id_column):\n",
    "    row = df.loc[df[id_column] == id]\n",
    "    return row"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [],
   "source": [
    "def extract_cell_value(row, col_index):\n",
    "    # Récupération de la valeur de la cellule\n",
    "    cell_value = row.iloc[0,col_index]\n",
    "    return cell_value"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/local/anaconda3/lib/python3.6/site-packages/IPython/core/interactiveshell.py:3057: DtypeWarning: Columns (9) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    }
   ],
   "source": [
    "### Chemin vers le fichier CSV contenant les données de génotypage Illumina\n",
    "GPL_file = \"newGPL18545_humanexome-12v1-1_a.csv\"\n",
    "GSE_file= \"GSE148375_processed_data.txt\"\n",
    "\n",
    "# Lecture des données de génotypage à partir du fichier CSV\n",
    "GPL = pd.read_csv(GPL_file, sep=\",\")\n",
    "GSE = pd.read_csv(GSE_file, sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [],
   "source": [
    "def ajouter_colonnes(df):\n",
    "    colonnes_a_ajouter = [f\"sample_{i}\" for i in range(1, num_samples+1)]\n",
    "    df[colonnes_a_ajouter] = pd.DataFrame([[np.nan]*num_samples], index=df.index)\n",
    "    return df\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [],
   "source": [
    "from tqdm import tqdm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID_REF</th>\n",
       "      <th>1900017</th>\n",
       "      <th>1900029</th>\n",
       "      <th>1900030</th>\n",
       "      <th>1900031</th>\n",
       "      <th>1900040</th>\n",
       "      <th>1900043</th>\n",
       "      <th>1900046</th>\n",
       "      <th>1900048</th>\n",
       "      <th>1900049</th>\n",
       "      <th>...</th>\n",
       "      <th>215606</th>\n",
       "      <th>215753</th>\n",
       "      <th>215762</th>\n",
       "      <th>215763</th>\n",
       "      <th>215764</th>\n",
       "      <th>215769</th>\n",
       "      <th>215771</th>\n",
       "      <th>215772</th>\n",
       "      <th>215773</th>\n",
       "      <th>215827</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>exm2268640-0_B_F_1984844585</td>\n",
       "      <td>AG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>...</td>\n",
       "      <td>GG</td>\n",
       "      <td>NC</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>exm41-0_B_F_1921435147</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>...</td>\n",
       "      <td>GG</td>\n",
       "      <td>AA</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>exm1916089-0_B_R_1927689775</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>...</td>\n",
       "      <td>GG</td>\n",
       "      <td>NC</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>exm44-0_B_R_1921538602</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>...</td>\n",
       "      <td>GG</td>\n",
       "      <td>AG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>exm46-0_T_F_1921333919</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>...</td>\n",
       "      <td>GG</td>\n",
       "      <td>NC</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 3400 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                        ID_REF 1900017 1900029 1900030 1900031 1900040  \\\n",
       "0  exm2268640-0_B_F_1984844585      AG      GG      GG      GG      GG   \n",
       "1       exm41-0_B_F_1921435147      GG      GG      GG      GG      GG   \n",
       "2  exm1916089-0_B_R_1927689775      GG      GG      GG      GG      GG   \n",
       "3       exm44-0_B_R_1921538602      GG      GG      GG      GG      GG   \n",
       "4       exm46-0_T_F_1921333919      GG      GG      GG      GG      GG   \n",
       "\n",
       "  1900043 1900046 1900048 1900049  ... 215606 215753 215762 215763 215764  \\\n",
       "0      GG      GG      GG      GG  ...     GG     NC     GG     GG     GG   \n",
       "1      GG      GG      GG      GG  ...     GG     AA     GG     GG     GG   \n",
       "2      GG      GG      GG      GG  ...     GG     NC     GG     GG     GG   \n",
       "3      GG      GG      GG      GG  ...     GG     AG     GG     GG     GG   \n",
       "4      GG      GG      GG      GG  ...     GG     NC     GG     GG     GG   \n",
       "\n",
       "  215769 215771 215772 215773 215827  \n",
       "0     GG     GG     GG     GG     GG  \n",
       "1     GG     GG     GG     GG     GG  \n",
       "2     GG     GG     GG     GG     GG  \n",
       "3     GG     GG     GG     GG     GG  \n",
       "4     GG     GG     GG     GG     GG  \n",
       "\n",
       "[5 rows x 3400 columns]"
      ]
     },
     "execution_count": 86,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "GSE.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [],
   "source": [
    "remove_ends_column(GPL, \"SNP\")\n",
    "\n",
    "# Conversion des génotypes au format VCF\n",
    "hapmap = pd.DataFrame(columns=[\"exm#\", \"alleles\", \"chrom\", \"pos\", \"strand\", \"assembly#\", \"center\",\n",
    "                               \"protLSID\", \"assayLSID\", \"panelLSID\", \"QCcode\"])\n",
    "\n",
    "# supprimer la première colonne de GPL\n",
    "GSE_prime=GSE.copy()\n",
    "GSE_prime.drop(GSE_prime.columns[0], axis=1, inplace=True)\n",
    "# insérer GPL à la fin de hapmap\n",
    "hapmap = pd.concat([hapmap, GSE_prime], ignore_index=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>exm#</th>\n",
       "      <th>alleles</th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos</th>\n",
       "      <th>strand</th>\n",
       "      <th>assembly#</th>\n",
       "      <th>center</th>\n",
       "      <th>protLSID</th>\n",
       "      <th>assayLSID</th>\n",
       "      <th>panelLSID</th>\n",
       "      <th>...</th>\n",
       "      <th>215606</th>\n",
       "      <th>215753</th>\n",
       "      <th>215762</th>\n",
       "      <th>215763</th>\n",
       "      <th>215764</th>\n",
       "      <th>215769</th>\n",
       "      <th>215771</th>\n",
       "      <th>215772</th>\n",
       "      <th>215773</th>\n",
       "      <th>215827</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>GG</td>\n",
       "      <td>NC</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>GG</td>\n",
       "      <td>AA</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>GG</td>\n",
       "      <td>NC</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>GG</td>\n",
       "      <td>AG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>GG</td>\n",
       "      <td>NC</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 3410 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  exm# alleles chrom  pos strand assembly# center protLSID assayLSID  \\\n",
       "0  NaN     NaN   NaN  NaN    NaN       NaN    NaN      NaN       NaN   \n",
       "1  NaN     NaN   NaN  NaN    NaN       NaN    NaN      NaN       NaN   \n",
       "2  NaN     NaN   NaN  NaN    NaN       NaN    NaN      NaN       NaN   \n",
       "3  NaN     NaN   NaN  NaN    NaN       NaN    NaN      NaN       NaN   \n",
       "4  NaN     NaN   NaN  NaN    NaN       NaN    NaN      NaN       NaN   \n",
       "\n",
       "  panelLSID  ... 215606 215753 215762 215763 215764 215769 215771 215772  \\\n",
       "0       NaN  ...     GG     NC     GG     GG     GG     GG     GG     GG   \n",
       "1       NaN  ...     GG     AA     GG     GG     GG     GG     GG     GG   \n",
       "2       NaN  ...     GG     NC     GG     GG     GG     GG     GG     GG   \n",
       "3       NaN  ...     GG     AG     GG     GG     GG     GG     GG     GG   \n",
       "4       NaN  ...     GG     NC     GG     GG     GG     GG     GG     GG   \n",
       "\n",
       "  215773 215827  \n",
       "0     GG     GG  \n",
       "1     GG     GG  \n",
       "2     GG     GG  \n",
       "3     GG     GG  \n",
       "4     GG     GG  \n",
       "\n",
       "[5 rows x 3410 columns]"
      ]
     },
     "execution_count": 81,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hapmap.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "GPL_prime=GPL.copy()\n",
    "row=extract_row_by_id(GPL_prime, \"exm2268640-0_B_F_1984844585\",\"IlmnID\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'exm2268640'"
      ]
     },
     "execution_count": 100,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "extract_cell_value(row,1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'exm2268640'"
      ]
     },
     "execution_count": 99,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "extract_cell_value(row,1)\n",
    "row.iloc[0,1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "\n",
      "  0%|          | 0/242901 [00:00<?, ?it/s]\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "  0%|          | 1/242901 [00:32<2218:04:05, 32.87s/it]\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "  0%|          | 2/242901 [01:04<2197:09:56, 32.56s/it]\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "  0%|          | 3/242901 [01:35<2170:01:13, 32.16s/it]\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "  0%|          | 4/242901 [02:07<2157:06:33, 31.97s/it]\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "  0%|          | 5/242901 [02:42<2223:11:56, 32.95s/it]\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "  0%|          | 6/242901 [03:17<2253:16:27, 33.40s/it]\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "  0%|          | 7/242901 [03:48<2212:17:29, 32.79s/it]\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "  0%|          | 8/242901 [04:19<2179:01:03, 32.30s/it]\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "  0%|          | 9/242901 [04:52<2196:54:50, 32.56s/it]\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "  0%|          | 10/242901 [05:25<2196:22:56, 32.55s/it]\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "  0%|          | 11/242901 [05:57<2190:19:18, 32.46s/it]\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "  0%|          | 12/242901 [06:29<2169:16:30, 32.15s/it]\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "  0%|          | 13/242901 [06:59<2141:43:38, 31.74s/it]\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "  0%|          | 14/242901 [07:30<2122:06:15, 31.45s/it]\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "  0%|          | 15/242901 [08:02<2136:12:17, 31.66s/it]\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "  0%|          | 16/242901 [08:37<2203:30:30, 32.66s/it]\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "  0%|          | 17/242901 [09:11<2231:17:07, 33.07s/it]\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "  0%|          | 18/242901 [09:48<2296:38:43, 34.04s/it]\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "  0%|          | 19/242901 [10:23<2329:38:19, 34.53s/it]\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "  0%|          | 20/242901 [11:09<2555:19:06, 37.88s/it]\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "  0%|          | 21/242901 [12:16<3152:25:31, 46.73s/it]\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "  0%|          | 22/242901 [13:25<3594:44:15, 53.28s/it]\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "  0%|          | 23/242901 [14:33<3899:23:15, 57.80s/it]\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "  0%|          | 24/242901 [15:39<4069:41:17, 60.32s/it]\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "  0%|          | 25/242901 [16:30<3876:01:03, 57.45s/it]\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "  0%|          | 26/242901 [17:37<4064:07:14, 60.24s/it]\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "  0%|          | 27/242901 [18:42<4170:18:57, 61.81s/it]\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "  0%|          | 28/242901 [19:48<4238:48:08, 62.83s/it]\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "  0%|          | 29/242901 [20:52<4277:26:06, 63.40s/it]\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "  0%|          | 30/242901 [21:29<3729:26:48, 55.28s/it]\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "  0%|          | 31/242901 [22:00<3240:26:06, 48.03s/it]\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "  0%|          | 32/242901 [22:31<2908:13:53, 43.11s/it]\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "  0%|          | 33/242901 [23:03<2684:24:24, 39.79s/it]\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "  0%|          | 34/242901 [23:35<2526:30:45, 37.45s/it]\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "  0%|          | 35/242901 [24:07<2400:58:28, 35.59s/it]\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "  0%|          | 36/242901 [24:38<2309:58:52, 34.24s/it]\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "  0%|          | 37/242901 [25:09<2249:30:12, 33.34s/it]\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "  0%|          | 38/242901 [25:40<2203:09:42, 32.66s/it]\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "  0%|          | 39/242901 [26:14<2234:58:36, 33.13s/it]\u001b[A\u001b[A\u001b[A"
     ]
    }
   ],
   "source": [
    "#hapmap = ajouter_colonnes(hapmap)\n",
    "\n",
    "for i in tqdm(range(0, len(GPL.index))):\n",
    "    # Conversion des génotypes en format VCF\n",
    "    hapmap.loc[i,\"exm#\"] = GPL.loc[i,\"IlmnID\"]\n",
    "    hapmap.loc[i, \"alleles\"] = GPL.loc[i,\"SNP\"]\n",
    "    hapmap.loc[i,\"chrom\"] = GPL.loc[i,\"Chr\"]\n",
    "    hapmap.loc[i,\"pos\"] = GPL.loc[i,\"MapInfo\"]\n",
    "    hapmap.loc[i,\"strand\"] = GPL.loc[i,\"RefStrand\"]\n",
    "    hapmap.loc[i,\"assembly#\"] = \"NA\"\n",
    "    hapmap.loc[i,\"center\"] = \"NA\"\n",
    "    hapmap.loc[i,\"protLSID\"] = \"NA\"\n",
    "    hapmap.loc[i,\"assayLSID\"] = \"NA\"\n",
    "    hapmap.loc[i,\"panelLSID\"] = \"NA\"\n",
    "    hapmap.loc[i,\"QCcode\"] = \"NA\"\n",
    "    for j in range (11,len(hapmap.columns)):\n",
    "        line=extract_row_by_id(GSE, GPL.loc[i,\"IlmnID\"], \"ID_REF\")\n",
    "        hapmap.iloc[i,j] = extract_cell_value(line,j-11)\n",
    "\n",
    "# Écriture des génotypes convertis en format VCF dans un fichier\n",
    "hapmap.to_csv(\"data.hapmap\", sep=\"\\t\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
