{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "2bc94276",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Defaulting to user installation because normal site-packages is not writeable\n",
      "Collecting dask\n",
      "  Downloading dask-2023.3.0-py3-none-any.whl (1.2 MB)\n",
      "\u001b[2K     \u001b[38;2;114;156;31m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m1.2/1.2 MB\u001b[0m \u001b[31m3.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0mm eta \u001b[36m0:00:01\u001b[0m[36m0:00:01\u001b[0m\n",
      "\u001b[?25hCollecting fsspec>=0.6.0\n",
      "  Downloading fsspec-2023.1.0-py3-none-any.whl (143 kB)\n",
      "\u001b[2K     \u001b[38;2;114;156;31m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m143.0/143.0 KB\u001b[0m \u001b[31m1.8 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m MB/s\u001b[0m eta \u001b[36m0:00:01\u001b[0m\n",
      "\u001b[?25hCollecting toolz>=0.8.2\n",
      "  Downloading toolz-0.12.0-py3-none-any.whl (55 kB)\n",
      "\u001b[2K     \u001b[38;2;114;156;31m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m55.8/55.8 KB\u001b[0m \u001b[31m1.0 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m[31m4.1 MB/s\u001b[0m eta \u001b[36m0:00:01\u001b[0m\n",
      "\u001b[?25hRequirement already satisfied: packaging>=20.0 in /home/legrand/.local/lib/python3.10/site-packages (from dask) (21.3)\n",
      "Requirement already satisfied: click>=7.0 in /usr/lib/python3/dist-packages (from dask) (8.0.3)\n",
      "Requirement already satisfied: pyyaml>=5.3.1 in /usr/lib/python3/dist-packages (from dask) (5.4.1)\n",
      "Collecting partd>=1.2.0\n",
      "  Downloading partd-1.3.0-py3-none-any.whl (18 kB)\n",
      "Collecting cloudpickle>=1.1.1\n",
      "  Downloading cloudpickle-2.2.1-py3-none-any.whl (25 kB)\n",
      "Requirement already satisfied: pyparsing!=3.0.5,>=2.0.2 in /usr/lib/python3/dist-packages (from packaging>=20.0->dask) (2.4.7)\n",
      "Collecting locket\n",
      "  Downloading locket-1.0.0-py2.py3-none-any.whl (4.4 kB)\n",
      "Installing collected packages: toolz, locket, fsspec, cloudpickle, partd, dask\n",
      "\u001b[33m  WARNING: The script dask is installed in '/home/legrand/.local/bin' which is not on PATH.\n",
      "  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.\u001b[0m\u001b[33m\n",
      "\u001b[0mSuccessfully installed cloudpickle-2.2.1 dask-2023.3.0 fsspec-2023.1.0 locket-1.0.0 partd-1.3.0 toolz-0.12.0\n"
     ]
    }
   ],
   "source": [
    "!pip install dask"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "efabeb95",
   "metadata": {},
   "outputs": [],
   "source": [
    "import matplotlib.pyplot as plt\n",
    "import dask.dataframe as dd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "dd808ab3",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = dd.read_csv('GSE148375_processed_data.txt', delimiter='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "76cf23f0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID_REF</th>\n",
       "      <th>1900017</th>\n",
       "      <th>1900029</th>\n",
       "      <th>1900030</th>\n",
       "      <th>1900031</th>\n",
       "      <th>1900040</th>\n",
       "      <th>1900043</th>\n",
       "      <th>1900046</th>\n",
       "      <th>1900048</th>\n",
       "      <th>1900049</th>\n",
       "      <th>...</th>\n",
       "      <th>215606</th>\n",
       "      <th>215753</th>\n",
       "      <th>215762</th>\n",
       "      <th>215763</th>\n",
       "      <th>215764</th>\n",
       "      <th>215769</th>\n",
       "      <th>215771</th>\n",
       "      <th>215772</th>\n",
       "      <th>215773</th>\n",
       "      <th>215827</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>exm2268640-0_B_F_1984844585</td>\n",
       "      <td>AG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>...</td>\n",
       "      <td>GG</td>\n",
       "      <td>NC</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>exm41-0_B_F_1921435147</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>...</td>\n",
       "      <td>GG</td>\n",
       "      <td>AA</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>exm1916089-0_B_R_1927689775</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>...</td>\n",
       "      <td>GG</td>\n",
       "      <td>NC</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>exm44-0_B_R_1921538602</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>...</td>\n",
       "      <td>GG</td>\n",
       "      <td>AG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>exm46-0_T_F_1921333919</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>...</td>\n",
       "      <td>GG</td>\n",
       "      <td>NC</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 3400 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                        ID_REF 1900017 1900029 1900030 1900031 1900040  \\\n",
       "0  exm2268640-0_B_F_1984844585      AG      GG      GG      GG      GG   \n",
       "1       exm41-0_B_F_1921435147      GG      GG      GG      GG      GG   \n",
       "2  exm1916089-0_B_R_1927689775      GG      GG      GG      GG      GG   \n",
       "3       exm44-0_B_R_1921538602      GG      GG      GG      GG      GG   \n",
       "4       exm46-0_T_F_1921333919      GG      GG      GG      GG      GG   \n",
       "\n",
       "  1900043 1900046 1900048 1900049  ... 215606 215753 215762 215763 215764  \\\n",
       "0      GG      GG      GG      GG  ...     GG     NC     GG     GG     GG   \n",
       "1      GG      GG      GG      GG  ...     GG     AA     GG     GG     GG   \n",
       "2      GG      GG      GG      GG  ...     GG     NC     GG     GG     GG   \n",
       "3      GG      GG      GG      GG  ...     GG     AG     GG     GG     GG   \n",
       "4      GG      GG      GG      GG  ...     GG     NC     GG     GG     GG   \n",
       "\n",
       "  215769 215771 215772 215773 215827  \n",
       "0     GG     GG     GG     GG     GG  \n",
       "1     GG     GG     GG     GG     GG  \n",
       "2     GG     GG     GG     GG     GG  \n",
       "3     GG     GG     GG     GG     GG  \n",
       "4     GG     GG     GG     GG     GG  \n",
       "\n",
       "[5 rows x 3400 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "68578af8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(Delayed('int-0fe50c31-3d90-4d72-8dc5-2b2abf80898e'), 3400)"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "9cb6a77f",
   "metadata": {},
   "outputs": [],
   "source": [
    "import csv\n",
    "\n",
    "# Ouverture du fichier CSV en mode lecture\n",
    "with open('GPL18545_humanexome-12v1-1_a.csv', 'r') as csvfile:\n",
    "    # Lecture du contenu du fichier CSV\n",
    "    csvreader = csv.reader(csvfile)\n",
    "\n",
    "    # Suppression des 7 premières lignes\n",
    "    for i in range(7):\n",
    "        next(csvreader)\n",
    "\n",
    "    # Ouverture du fichier CSV en mode écriture\n",
    "    with open('nouveau_fichier.csv', 'w', newline='') as newcsvfile:\n",
    "        # Écriture du contenu du fichier CSV sans les 7 premières lignes\n",
    "        csvwriter = csv.writer(newcsvfile)\n",
    "        for row in csvreader:\n",
    "            csvwriter.writerow(row)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "a013ed50",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "cc5ed25b",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_14728/1554040134.py:2: DtypeWarning: Columns (9) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  genotypes = pd.read_csv(input_file, sep=\",\")\n"
     ]
    }
   ],
   "source": [
    "input_file = \"nouveau_fichier.csv\"\n",
    "genotypes = pd.read_csv(input_file, sep=\",\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "afc786c6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>IlmnID</th>\n",
       "      <th>Name</th>\n",
       "      <th>IlmnStrand</th>\n",
       "      <th>SNP</th>\n",
       "      <th>AddressA_ID</th>\n",
       "      <th>AlleleA_ProbeSeq</th>\n",
       "      <th>AddressB_ID</th>\n",
       "      <th>AlleleB_ProbeSeq</th>\n",
       "      <th>GenomeBuild</th>\n",
       "      <th>Chr</th>\n",
       "      <th>...</th>\n",
       "      <th>Ploidy</th>\n",
       "      <th>Species</th>\n",
       "      <th>Source</th>\n",
       "      <th>SourceVersion</th>\n",
       "      <th>SourceStrand</th>\n",
       "      <th>SourceSeq</th>\n",
       "      <th>TopGenomicSeq</th>\n",
       "      <th>BeadSetID</th>\n",
       "      <th>Exp_Clusters</th>\n",
       "      <th>RefStrand</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>exm-IND1-200449980-0_M_R_1990486447</td>\n",
       "      <td>exm-IND1-200449980</td>\n",
       "      <td>MINUS</td>\n",
       "      <td>[D/I]</td>\n",
       "      <td>69606251.0</td>\n",
       "      <td>CTGTGGAACTTTCCGGAGCTCGGCACCCTTCCCCCAAGCCTACCTG...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>37.1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>diploid</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>1000genomes</td>\n",
       "      <td>0.0</td>\n",
       "      <td>PLUS</td>\n",
       "      <td>GAGGGCCGCTCAGCGAGGGCGGGACAGAACCTCTCCCGGGCTGGGA...</td>\n",
       "      <td>GAGGGCCGCTCAGCGAGGGCGGGACAGAACCTCTCCCGGGCTGGGA...</td>\n",
       "      <td>662.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>exm-IND1-201453487-0_M_R_1990486448</td>\n",
       "      <td>exm-IND1-201453487</td>\n",
       "      <td>MINUS</td>\n",
       "      <td>[D/I]</td>\n",
       "      <td>3775555.0</td>\n",
       "      <td>TCCTGCAACCAGGGCCGATACCCCCTCATCCAGACGCTACGGCAGG...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>37.1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>diploid</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>1000genomes</td>\n",
       "      <td>0.0</td>\n",
       "      <td>PLUS</td>\n",
       "      <td>CATGGAGCTGTGTTTGTACCCCACCTCCAAATTCCACCACTGGCCC...</td>\n",
       "      <td>CATGGAGCTGTGTTTGTACCCCACCTCCAAATTCCACCACTGGCCC...</td>\n",
       "      <td>662.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>exm-IND1-85310248-0_P_F_1990486624</td>\n",
       "      <td>exm-IND1-85310248</td>\n",
       "      <td>PLUS</td>\n",
       "      <td>[I/D]</td>\n",
       "      <td>56795122.0</td>\n",
       "      <td>TCAGAACCATGGCTCCAAAACAAAAGAAAAAGACATCACGTGGCAA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>37.1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>diploid</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>1000genomes</td>\n",
       "      <td>0.0</td>\n",
       "      <td>PLUS</td>\n",
       "      <td>CTTTATGCAGCCCAAGTCAGAACCATGGCTCCAAAACAAAAGAAAA...</td>\n",
       "      <td>CTTTATGCAGCCCAAGTCAGAACCATGGCTCCAAAACAAAAGAAAA...</td>\n",
       "      <td>688.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>exm-IND10-102817747-0_P_F_1990486569</td>\n",
       "      <td>exm-IND10-102817747</td>\n",
       "      <td>PLUS</td>\n",
       "      <td>[I/D]</td>\n",
       "      <td>30777118.0</td>\n",
       "      <td>CTCTGCAAGTTTTGTGCCTTAGCGGAAAGAGGCGACATCTCTGGCA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>37.1</td>\n",
       "      <td>10</td>\n",
       "      <td>...</td>\n",
       "      <td>diploid</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>1000genomes</td>\n",
       "      <td>0.0</td>\n",
       "      <td>PLUS</td>\n",
       "      <td>GGGAACACCTGGCTCTGCAAGTTTTGTGCCTTAGCGGAAAGAGGCG...</td>\n",
       "      <td>GGGAACACCTGGCTCTGCAAGTTTTGTGCCTTAGCGGAAAGAGGCG...</td>\n",
       "      <td>662.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>exm-IND10-18329639-0_M_R_1990486442</td>\n",
       "      <td>exm-IND10-18329639</td>\n",
       "      <td>MINUS</td>\n",
       "      <td>[D/I]</td>\n",
       "      <td>87684402.0</td>\n",
       "      <td>AAGGCTGCTCCTATGGCTAGGCCATCTGCAAAATTATGCAGGCTGT...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>37.1</td>\n",
       "      <td>10</td>\n",
       "      <td>...</td>\n",
       "      <td>diploid</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>1000genomes</td>\n",
       "      <td>0.0</td>\n",
       "      <td>PLUS</td>\n",
       "      <td>CTGTGTTTATTTGGTTCCTTAGGTAAAGCCATTAGCTTGTTAGCAA...</td>\n",
       "      <td>CTGTGTTTATTTGGTTCCTTAGGTAAAGCCATTAGCTTGTTAGCAA...</td>\n",
       "      <td>662.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                 IlmnID                 Name IlmnStrand  \\\n",
       "0   exm-IND1-200449980-0_M_R_1990486447   exm-IND1-200449980      MINUS   \n",
       "1   exm-IND1-201453487-0_M_R_1990486448   exm-IND1-201453487      MINUS   \n",
       "2    exm-IND1-85310248-0_P_F_1990486624    exm-IND1-85310248       PLUS   \n",
       "3  exm-IND10-102817747-0_P_F_1990486569  exm-IND10-102817747       PLUS   \n",
       "4   exm-IND10-18329639-0_M_R_1990486442   exm-IND10-18329639      MINUS   \n",
       "\n",
       "     SNP  AddressA_ID                                   AlleleA_ProbeSeq  \\\n",
       "0  [D/I]   69606251.0  CTGTGGAACTTTCCGGAGCTCGGCACCCTTCCCCCAAGCCTACCTG...   \n",
       "1  [D/I]    3775555.0  TCCTGCAACCAGGGCCGATACCCCCTCATCCAGACGCTACGGCAGG...   \n",
       "2  [I/D]   56795122.0  TCAGAACCATGGCTCCAAAACAAAAGAAAAAGACATCACGTGGCAA...   \n",
       "3  [I/D]   30777118.0  CTCTGCAAGTTTTGTGCCTTAGCGGAAAGAGGCGACATCTCTGGCA...   \n",
       "4  [D/I]   87684402.0  AAGGCTGCTCCTATGGCTAGGCCATCTGCAAAATTATGCAGGCTGT...   \n",
       "\n",
       "   AddressB_ID AlleleB_ProbeSeq  GenomeBuild Chr  ...   Ploidy       Species  \\\n",
       "0          NaN              NaN         37.1   1  ...  diploid  Homo sapiens   \n",
       "1          NaN              NaN         37.1   1  ...  diploid  Homo sapiens   \n",
       "2          NaN              NaN         37.1   1  ...  diploid  Homo sapiens   \n",
       "3          NaN              NaN         37.1  10  ...  diploid  Homo sapiens   \n",
       "4          NaN              NaN         37.1  10  ...  diploid  Homo sapiens   \n",
       "\n",
       "        Source SourceVersion  SourceStrand  \\\n",
       "0  1000genomes           0.0          PLUS   \n",
       "1  1000genomes           0.0          PLUS   \n",
       "2  1000genomes           0.0          PLUS   \n",
       "3  1000genomes           0.0          PLUS   \n",
       "4  1000genomes           0.0          PLUS   \n",
       "\n",
       "                                           SourceSeq  \\\n",
       "0  GAGGGCCGCTCAGCGAGGGCGGGACAGAACCTCTCCCGGGCTGGGA...   \n",
       "1  CATGGAGCTGTGTTTGTACCCCACCTCCAAATTCCACCACTGGCCC...   \n",
       "2  CTTTATGCAGCCCAAGTCAGAACCATGGCTCCAAAACAAAAGAAAA...   \n",
       "3  GGGAACACCTGGCTCTGCAAGTTTTGTGCCTTAGCGGAAAGAGGCG...   \n",
       "4  CTGTGTTTATTTGGTTCCTTAGGTAAAGCCATTAGCTTGTTAGCAA...   \n",
       "\n",
       "                                       TopGenomicSeq BeadSetID  Exp_Clusters  \\\n",
       "0  GAGGGCCGCTCAGCGAGGGCGGGACAGAACCTCTCCCGGGCTGGGA...     662.0           3.0   \n",
       "1  CATGGAGCTGTGTTTGTACCCCACCTCCAAATTCCACCACTGGCCC...     662.0           3.0   \n",
       "2  CTTTATGCAGCCCAAGTCAGAACCATGGCTCCAAAACAAAAGAAAA...     688.0           3.0   \n",
       "3  GGGAACACCTGGCTCTGCAAGTTTTGTGCCTTAGCGGAAAGAGGCG...     662.0           3.0   \n",
       "4  CTGTGTTTATTTGGTTCCTTAGGTAAAGCCATTAGCTTGTTAGCAA...     662.0           3.0   \n",
       "\n",
       "   RefStrand  \n",
       "0          -  \n",
       "1          -  \n",
       "2          +  \n",
       "3          +  \n",
       "4          -  \n",
       "\n",
       "[5 rows x 21 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genotypes.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "c2172f88",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_14728/1736392596.py:5: DtypeWarning: Columns (9) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  genotypes = pd.read_csv(input_file, sep=\",\")\n"
     ]
    }
   ],
   "source": [
    "# Chemin vers le fichier CSV contenant les données de génotypage Illumina\n",
    "input_file = \"nouveau_fichier.csv\"\n",
    "\n",
    "# Lecture des données de génotypage à partir du fichier CSV\n",
    "genotypes = pd.read_csv(input_file, sep=\",\")\n",
    "\n",
    "# Conversion des génotypes au format VCF\n",
    "vcf = pd.DataFrame(columns=[\"CHROM\", \"POS\", \"ID\", \"REF\", \"ALT\", \"QUAL\", \"FILTER\", \"INFO\", \"FORMAT\"])\n",
    "\n",
    "vcf[\"CHROM\"] = genotypes[\"Chr\"]\n",
    "vcf[\"POS\"] = genotypes[\"MapInfo\"]\n",
    "vcf[\"ID\"] = genotypes[\"IlmnID\"]\n",
    "vcf[\"REF\"] = genotypes[\"AlleleA_ProbeSeq\"]\n",
    "vcf[\"ALT\"] = genotypes[\"AlleleB_ProbeSeq\"]\n",
    "vcf[\"QUAL\"] = \".\"\n",
    "vcf[\"FILTER\"] = \".\"\n",
    "vcf[\"INFO\"] = \".\"\n",
    "vcf[\"FORMAT\"] = \"GT\"\n",
    "\n",
    "for i in range(0, len(genotypes.index)):\n",
    "    # Conversion des génotypes en format VCF\n",
    "    genotype = genotypes.iloc[i, 6:]\n",
    "    vcf.loc[i, \"SAMPLE\"] = \"Sample_\" + str(i + 1)\n",
    "    vcf.loc[i, \"SAMPLE\"] = \":\".join([str(x) for x in genotype])\n",
    "\n",
    "# Écriture des génotypes convertis en format VCF dans un fichier\n",
    "vcf.to_csv(\"genotypes.vcf\", sep=\"\\t\", index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "e1ce5b44",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_14728/913550925.py:2: DtypeWarning: Columns (0) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  genotypevcf = pd.read_csv(vcf_file, sep=\"\\t\")\n"
     ]
    }
   ],
   "source": [
    "vcf_file = \"genotypes.vcf\"\n",
    "genotypevcf = pd.read_csv(vcf_file, sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "f4b88b5b",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>202183358.0</td>\n",
       "      <td>exm-IND1-200449980-0_M_R_1990486447</td>\n",
       "      <td>CTGTGGAACTTTCCGGAGCTCGGCACCCTTCCCCCAAGCCTACCTG...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>GT</td>\n",
       "      <td>nan:nan:37.1:1:202183358.0:diploid:Homo sapien...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>203186865.0</td>\n",
       "      <td>exm-IND1-201453487-0_M_R_1990486448</td>\n",
       "      <td>TCCTGCAACCAGGGCCGATACCCCCTCATCCAGACGCTACGGCAGG...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>GT</td>\n",
       "      <td>nan:nan:37.1:1:203186865.0:diploid:Homo sapien...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>85537661.0</td>\n",
       "      <td>exm-IND1-85310248-0_P_F_1990486624</td>\n",
       "      <td>TCAGAACCATGGCTCCAAAACAAAAGAAAAAGACATCACGTGGCAA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>GT</td>\n",
       "      <td>nan:nan:37.1:1:85537661.0:diploid:Homo sapiens...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10</td>\n",
       "      <td>102827758.0</td>\n",
       "      <td>exm-IND10-102817747-0_P_F_1990486569</td>\n",
       "      <td>CTCTGCAAGTTTTGTGCCTTAGCGGAAAGAGGCGACATCTCTGGCA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>GT</td>\n",
       "      <td>nan:nan:37.1:10:102827758.0:diploid:Homo sapie...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>10</td>\n",
       "      <td>18289634.0</td>\n",
       "      <td>exm-IND10-18329639-0_M_R_1990486442</td>\n",
       "      <td>AAGGCTGCTCCTATGGCTAGGCCATCTGCAAAATTATGCAGGCTGT...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>GT</td>\n",
       "      <td>nan:nan:37.1:10:18289634.0:diploid:Homo sapien...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM          POS                                    ID  \\\n",
       "0     1  202183358.0   exm-IND1-200449980-0_M_R_1990486447   \n",
       "1     1  203186865.0   exm-IND1-201453487-0_M_R_1990486448   \n",
       "2     1   85537661.0    exm-IND1-85310248-0_P_F_1990486624   \n",
       "3    10  102827758.0  exm-IND10-102817747-0_P_F_1990486569   \n",
       "4    10   18289634.0   exm-IND10-18329639-0_M_R_1990486442   \n",
       "\n",
       "                                                 REF  ALT QUAL FILTER INFO  \\\n",
       "0  CTGTGGAACTTTCCGGAGCTCGGCACCCTTCCCCCAAGCCTACCTG...  NaN    .      .    .   \n",
       "1  TCCTGCAACCAGGGCCGATACCCCCTCATCCAGACGCTACGGCAGG...  NaN    .      .    .   \n",
       "2  TCAGAACCATGGCTCCAAAACAAAAGAAAAAGACATCACGTGGCAA...  NaN    .      .    .   \n",
       "3  CTCTGCAAGTTTTGTGCCTTAGCGGAAAGAGGCGACATCTCTGGCA...  NaN    .      .    .   \n",
       "4  AAGGCTGCTCCTATGGCTAGGCCATCTGCAAAATTATGCAGGCTGT...  NaN    .      .    .   \n",
       "\n",
       "  FORMAT                                             SAMPLE  \n",
       "0     GT  nan:nan:37.1:1:202183358.0:diploid:Homo sapien...  \n",
       "1     GT  nan:nan:37.1:1:203186865.0:diploid:Homo sapien...  \n",
       "2     GT  nan:nan:37.1:1:85537661.0:diploid:Homo sapiens...  \n",
       "3     GT  nan:nan:37.1:10:102827758.0:diploid:Homo sapie...  \n",
       "4     GT  nan:nan:37.1:10:18289634.0:diploid:Homo sapien...  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genotypevcf.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2254c4e6",
   "metadata": {},
   "outputs": [],
   "source": [
    "file"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
