{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Chemin vers le fichier CSV\n",
    "filename = \"newGPL18545_humanexome-12v1-1_a.csv\"\n",
    "\n",
    "# Charger le fichier CSV dans un dataframe pandas\n",
    "df = pd.read_csv(filename)\n",
    "\n",
    "# Supprimer les 23 dernières lignes\n",
    "df = df.iloc[:-1]\n",
    "\n",
    "# Écrire les données modifiées dans le fichier CSV\n",
    "df.to_csv(\"newGPL18545_humanexome-12v1-1_a.csv\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Une fonction qui supprime les crochéts\n",
    "def remove_ends(input_string):\n",
    "    if len(input_string) < 3:\n",
    "        return \"input_string\"\n",
    "    else:\n",
    "        return input_string[1:-1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "metadata": {},
   "outputs": [],
   "source": [
    "def remove_ends_column(df, column_name):\n",
    "    # Créer une nouvelle colonne avec les valeurs modifiées\n",
    "    new_column = []\n",
    "    for i in df.index:\n",
    "        value = df.loc[i, column_name]\n",
    "        if isinstance(value, str):\n",
    "            new_value = remove_ends(value)\n",
    "        else:\n",
    "            new_value = \"\"\n",
    "        new_column.append(new_value)\n",
    "    # Remplacer la colonne existante par la nouvelle colonne\n",
    "    df[column_name] = new_column\n",
    "    return df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/local/anaconda3/lib/python3.6/site-packages/IPython/core/interactiveshell.py:3057: DtypeWarning: Columns (9) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    }
   ],
   "source": [
    "### Chemin vers le fichier CSV contenant les données de génotypage Illumina\n",
    "GPL_file = \"newGPL18545_humanexome-12v1-1_a.csv\"\n",
    "GSE_file= \"GSE148375_processed_data.txt\"\n",
    "\n",
    "# Lecture des données de génotypage à partir du fichier CSV\n",
    "GPL = pd.read_csv(GPL_file, sep=\",\")\n",
    "GSE = pd.read_csv(GSE_file, sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>IlmnID</th>\n",
       "      <th>Name</th>\n",
       "      <th>IlmnStrand</th>\n",
       "      <th>SNP</th>\n",
       "      <th>AddressA_ID</th>\n",
       "      <th>AlleleA_ProbeSeq</th>\n",
       "      <th>AddressB_ID</th>\n",
       "      <th>AlleleB_ProbeSeq</th>\n",
       "      <th>GenomeBuild</th>\n",
       "      <th>Chr</th>\n",
       "      <th>...</th>\n",
       "      <th>Ploidy</th>\n",
       "      <th>Species</th>\n",
       "      <th>Source</th>\n",
       "      <th>SourceVersion</th>\n",
       "      <th>SourceStrand</th>\n",
       "      <th>SourceSeq</th>\n",
       "      <th>TopGenomicSeq</th>\n",
       "      <th>BeadSetID</th>\n",
       "      <th>Exp_Clusters</th>\n",
       "      <th>RefStrand</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>exm-IND1-200449980-0_M_R_1990486447</td>\n",
       "      <td>exm-IND1-200449980</td>\n",
       "      <td>MINUS</td>\n",
       "      <td>[D/I]</td>\n",
       "      <td>69606251.0</td>\n",
       "      <td>CTGTGGAACTTTCCGGAGCTCGGCACCCTTCCCCCAAGCCTACCTG...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>37.1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>diploid</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>1000genomes</td>\n",
       "      <td>0.0</td>\n",
       "      <td>PLUS</td>\n",
       "      <td>GAGGGCCGCTCAGCGAGGGCGGGACAGAACCTCTCCCGGGCTGGGA...</td>\n",
       "      <td>GAGGGCCGCTCAGCGAGGGCGGGACAGAACCTCTCCCGGGCTGGGA...</td>\n",
       "      <td>662.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>exm-IND1-201453487-0_M_R_1990486448</td>\n",
       "      <td>exm-IND1-201453487</td>\n",
       "      <td>MINUS</td>\n",
       "      <td>[D/I]</td>\n",
       "      <td>3775555.0</td>\n",
       "      <td>TCCTGCAACCAGGGCCGATACCCCCTCATCCAGACGCTACGGCAGG...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>37.1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>diploid</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>1000genomes</td>\n",
       "      <td>0.0</td>\n",
       "      <td>PLUS</td>\n",
       "      <td>CATGGAGCTGTGTTTGTACCCCACCTCCAAATTCCACCACTGGCCC...</td>\n",
       "      <td>CATGGAGCTGTGTTTGTACCCCACCTCCAAATTCCACCACTGGCCC...</td>\n",
       "      <td>662.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>exm-IND1-85310248-0_P_F_1990486624</td>\n",
       "      <td>exm-IND1-85310248</td>\n",
       "      <td>PLUS</td>\n",
       "      <td>[I/D]</td>\n",
       "      <td>56795122.0</td>\n",
       "      <td>TCAGAACCATGGCTCCAAAACAAAAGAAAAAGACATCACGTGGCAA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>37.1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>diploid</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>1000genomes</td>\n",
       "      <td>0.0</td>\n",
       "      <td>PLUS</td>\n",
       "      <td>CTTTATGCAGCCCAAGTCAGAACCATGGCTCCAAAACAAAAGAAAA...</td>\n",
       "      <td>CTTTATGCAGCCCAAGTCAGAACCATGGCTCCAAAACAAAAGAAAA...</td>\n",
       "      <td>688.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>exm-IND10-102817747-0_P_F_1990486569</td>\n",
       "      <td>exm-IND10-102817747</td>\n",
       "      <td>PLUS</td>\n",
       "      <td>[I/D]</td>\n",
       "      <td>30777118.0</td>\n",
       "      <td>CTCTGCAAGTTTTGTGCCTTAGCGGAAAGAGGCGACATCTCTGGCA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>37.1</td>\n",
       "      <td>10</td>\n",
       "      <td>...</td>\n",
       "      <td>diploid</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>1000genomes</td>\n",
       "      <td>0.0</td>\n",
       "      <td>PLUS</td>\n",
       "      <td>GGGAACACCTGGCTCTGCAAGTTTTGTGCCTTAGCGGAAAGAGGCG...</td>\n",
       "      <td>GGGAACACCTGGCTCTGCAAGTTTTGTGCCTTAGCGGAAAGAGGCG...</td>\n",
       "      <td>662.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>exm-IND10-18329639-0_M_R_1990486442</td>\n",
       "      <td>exm-IND10-18329639</td>\n",
       "      <td>MINUS</td>\n",
       "      <td>[D/I]</td>\n",
       "      <td>87684402.0</td>\n",
       "      <td>AAGGCTGCTCCTATGGCTAGGCCATCTGCAAAATTATGCAGGCTGT...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>37.1</td>\n",
       "      <td>10</td>\n",
       "      <td>...</td>\n",
       "      <td>diploid</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>1000genomes</td>\n",
       "      <td>0.0</td>\n",
       "      <td>PLUS</td>\n",
       "      <td>CTGTGTTTATTTGGTTCCTTAGGTAAAGCCATTAGCTTGTTAGCAA...</td>\n",
       "      <td>CTGTGTTTATTTGGTTCCTTAGGTAAAGCCATTAGCTTGTTAGCAA...</td>\n",
       "      <td>662.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                 IlmnID                 Name IlmnStrand  \\\n",
       "0   exm-IND1-200449980-0_M_R_1990486447   exm-IND1-200449980      MINUS   \n",
       "1   exm-IND1-201453487-0_M_R_1990486448   exm-IND1-201453487      MINUS   \n",
       "2    exm-IND1-85310248-0_P_F_1990486624    exm-IND1-85310248       PLUS   \n",
       "3  exm-IND10-102817747-0_P_F_1990486569  exm-IND10-102817747       PLUS   \n",
       "4   exm-IND10-18329639-0_M_R_1990486442   exm-IND10-18329639      MINUS   \n",
       "\n",
       "     SNP  AddressA_ID                                   AlleleA_ProbeSeq  \\\n",
       "0  [D/I]   69606251.0  CTGTGGAACTTTCCGGAGCTCGGCACCCTTCCCCCAAGCCTACCTG...   \n",
       "1  [D/I]    3775555.0  TCCTGCAACCAGGGCCGATACCCCCTCATCCAGACGCTACGGCAGG...   \n",
       "2  [I/D]   56795122.0  TCAGAACCATGGCTCCAAAACAAAAGAAAAAGACATCACGTGGCAA...   \n",
       "3  [I/D]   30777118.0  CTCTGCAAGTTTTGTGCCTTAGCGGAAAGAGGCGACATCTCTGGCA...   \n",
       "4  [D/I]   87684402.0  AAGGCTGCTCCTATGGCTAGGCCATCTGCAAAATTATGCAGGCTGT...   \n",
       "\n",
       "   AddressB_ID AlleleB_ProbeSeq  GenomeBuild Chr  ...   Ploidy       Species  \\\n",
       "0          NaN              NaN         37.1   1  ...  diploid  Homo sapiens   \n",
       "1          NaN              NaN         37.1   1  ...  diploid  Homo sapiens   \n",
       "2          NaN              NaN         37.1   1  ...  diploid  Homo sapiens   \n",
       "3          NaN              NaN         37.1  10  ...  diploid  Homo sapiens   \n",
       "4          NaN              NaN         37.1  10  ...  diploid  Homo sapiens   \n",
       "\n",
       "        Source SourceVersion  SourceStrand  \\\n",
       "0  1000genomes           0.0          PLUS   \n",
       "1  1000genomes           0.0          PLUS   \n",
       "2  1000genomes           0.0          PLUS   \n",
       "3  1000genomes           0.0          PLUS   \n",
       "4  1000genomes           0.0          PLUS   \n",
       "\n",
       "                                           SourceSeq  \\\n",
       "0  GAGGGCCGCTCAGCGAGGGCGGGACAGAACCTCTCCCGGGCTGGGA...   \n",
       "1  CATGGAGCTGTGTTTGTACCCCACCTCCAAATTCCACCACTGGCCC...   \n",
       "2  CTTTATGCAGCCCAAGTCAGAACCATGGCTCCAAAACAAAAGAAAA...   \n",
       "3  GGGAACACCTGGCTCTGCAAGTTTTGTGCCTTAGCGGAAAGAGGCG...   \n",
       "4  CTGTGTTTATTTGGTTCCTTAGGTAAAGCCATTAGCTTGTTAGCAA...   \n",
       "\n",
       "                                       TopGenomicSeq BeadSetID  Exp_Clusters  \\\n",
       "0  GAGGGCCGCTCAGCGAGGGCGGGACAGAACCTCTCCCGGGCTGGGA...     662.0           3.0   \n",
       "1  CATGGAGCTGTGTTTGTACCCCACCTCCAAATTCCACCACTGGCCC...     662.0           3.0   \n",
       "2  CTTTATGCAGCCCAAGTCAGAACCATGGCTCCAAAACAAAAGAAAA...     688.0           3.0   \n",
       "3  GGGAACACCTGGCTCTGCAAGTTTTGTGCCTTAGCGGAAAGAGGCG...     662.0           3.0   \n",
       "4  CTGTGTTTATTTGGTTCCTTAGGTAAAGCCATTAGCTTGTTAGCAA...     662.0           3.0   \n",
       "\n",
       "   RefStrand  \n",
       "0          -  \n",
       "1          -  \n",
       "2          +  \n",
       "3          +  \n",
       "4          -  \n",
       "\n",
       "[5 rows x 21 columns]"
      ]
     },
     "execution_count": 138,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "GPL.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [],
   "source": [
    "from tqdm import tqdm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 139,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "GPL_prime=GPL.copy()\n",
    "GPL_prime=remove_ends_column(GPL_prime, \"SNP\")\n",
    "#row=extract_row_by_id(GPL_prime, \"exm2268640-0_B_F_1984844585\",\"IlmnID\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 140,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>IlmnID</th>\n",
       "      <th>Name</th>\n",
       "      <th>IlmnStrand</th>\n",
       "      <th>SNP</th>\n",
       "      <th>AddressA_ID</th>\n",
       "      <th>AlleleA_ProbeSeq</th>\n",
       "      <th>AddressB_ID</th>\n",
       "      <th>AlleleB_ProbeSeq</th>\n",
       "      <th>GenomeBuild</th>\n",
       "      <th>Chr</th>\n",
       "      <th>...</th>\n",
       "      <th>Ploidy</th>\n",
       "      <th>Species</th>\n",
       "      <th>Source</th>\n",
       "      <th>SourceVersion</th>\n",
       "      <th>SourceStrand</th>\n",
       "      <th>SourceSeq</th>\n",
       "      <th>TopGenomicSeq</th>\n",
       "      <th>BeadSetID</th>\n",
       "      <th>Exp_Clusters</th>\n",
       "      <th>RefStrand</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>exm-IND1-200449980-0_M_R_1990486447</td>\n",
       "      <td>exm-IND1-200449980</td>\n",
       "      <td>MINUS</td>\n",
       "      <td>D/I</td>\n",
       "      <td>69606251.0</td>\n",
       "      <td>CTGTGGAACTTTCCGGAGCTCGGCACCCTTCCCCCAAGCCTACCTG...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>37.1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>diploid</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>1000genomes</td>\n",
       "      <td>0.0</td>\n",
       "      <td>PLUS</td>\n",
       "      <td>GAGGGCCGCTCAGCGAGGGCGGGACAGAACCTCTCCCGGGCTGGGA...</td>\n",
       "      <td>GAGGGCCGCTCAGCGAGGGCGGGACAGAACCTCTCCCGGGCTGGGA...</td>\n",
       "      <td>662.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>exm-IND1-201453487-0_M_R_1990486448</td>\n",
       "      <td>exm-IND1-201453487</td>\n",
       "      <td>MINUS</td>\n",
       "      <td>D/I</td>\n",
       "      <td>3775555.0</td>\n",
       "      <td>TCCTGCAACCAGGGCCGATACCCCCTCATCCAGACGCTACGGCAGG...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>37.1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>diploid</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>1000genomes</td>\n",
       "      <td>0.0</td>\n",
       "      <td>PLUS</td>\n",
       "      <td>CATGGAGCTGTGTTTGTACCCCACCTCCAAATTCCACCACTGGCCC...</td>\n",
       "      <td>CATGGAGCTGTGTTTGTACCCCACCTCCAAATTCCACCACTGGCCC...</td>\n",
       "      <td>662.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>exm-IND1-85310248-0_P_F_1990486624</td>\n",
       "      <td>exm-IND1-85310248</td>\n",
       "      <td>PLUS</td>\n",
       "      <td>I/D</td>\n",
       "      <td>56795122.0</td>\n",
       "      <td>TCAGAACCATGGCTCCAAAACAAAAGAAAAAGACATCACGTGGCAA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>37.1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>diploid</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>1000genomes</td>\n",
       "      <td>0.0</td>\n",
       "      <td>PLUS</td>\n",
       "      <td>CTTTATGCAGCCCAAGTCAGAACCATGGCTCCAAAACAAAAGAAAA...</td>\n",
       "      <td>CTTTATGCAGCCCAAGTCAGAACCATGGCTCCAAAACAAAAGAAAA...</td>\n",
       "      <td>688.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>exm-IND10-102817747-0_P_F_1990486569</td>\n",
       "      <td>exm-IND10-102817747</td>\n",
       "      <td>PLUS</td>\n",
       "      <td>I/D</td>\n",
       "      <td>30777118.0</td>\n",
       "      <td>CTCTGCAAGTTTTGTGCCTTAGCGGAAAGAGGCGACATCTCTGGCA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>37.1</td>\n",
       "      <td>10</td>\n",
       "      <td>...</td>\n",
       "      <td>diploid</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>1000genomes</td>\n",
       "      <td>0.0</td>\n",
       "      <td>PLUS</td>\n",
       "      <td>GGGAACACCTGGCTCTGCAAGTTTTGTGCCTTAGCGGAAAGAGGCG...</td>\n",
       "      <td>GGGAACACCTGGCTCTGCAAGTTTTGTGCCTTAGCGGAAAGAGGCG...</td>\n",
       "      <td>662.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>exm-IND10-18329639-0_M_R_1990486442</td>\n",
       "      <td>exm-IND10-18329639</td>\n",
       "      <td>MINUS</td>\n",
       "      <td>D/I</td>\n",
       "      <td>87684402.0</td>\n",
       "      <td>AAGGCTGCTCCTATGGCTAGGCCATCTGCAAAATTATGCAGGCTGT...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>37.1</td>\n",
       "      <td>10</td>\n",
       "      <td>...</td>\n",
       "      <td>diploid</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>1000genomes</td>\n",
       "      <td>0.0</td>\n",
       "      <td>PLUS</td>\n",
       "      <td>CTGTGTTTATTTGGTTCCTTAGGTAAAGCCATTAGCTTGTTAGCAA...</td>\n",
       "      <td>CTGTGTTTATTTGGTTCCTTAGGTAAAGCCATTAGCTTGTTAGCAA...</td>\n",
       "      <td>662.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                 IlmnID                 Name IlmnStrand  SNP  \\\n",
       "0   exm-IND1-200449980-0_M_R_1990486447   exm-IND1-200449980      MINUS  D/I   \n",
       "1   exm-IND1-201453487-0_M_R_1990486448   exm-IND1-201453487      MINUS  D/I   \n",
       "2    exm-IND1-85310248-0_P_F_1990486624    exm-IND1-85310248       PLUS  I/D   \n",
       "3  exm-IND10-102817747-0_P_F_1990486569  exm-IND10-102817747       PLUS  I/D   \n",
       "4   exm-IND10-18329639-0_M_R_1990486442   exm-IND10-18329639      MINUS  D/I   \n",
       "\n",
       "   AddressA_ID                                   AlleleA_ProbeSeq  \\\n",
       "0   69606251.0  CTGTGGAACTTTCCGGAGCTCGGCACCCTTCCCCCAAGCCTACCTG...   \n",
       "1    3775555.0  TCCTGCAACCAGGGCCGATACCCCCTCATCCAGACGCTACGGCAGG...   \n",
       "2   56795122.0  TCAGAACCATGGCTCCAAAACAAAAGAAAAAGACATCACGTGGCAA...   \n",
       "3   30777118.0  CTCTGCAAGTTTTGTGCCTTAGCGGAAAGAGGCGACATCTCTGGCA...   \n",
       "4   87684402.0  AAGGCTGCTCCTATGGCTAGGCCATCTGCAAAATTATGCAGGCTGT...   \n",
       "\n",
       "   AddressB_ID AlleleB_ProbeSeq  GenomeBuild Chr  ...   Ploidy       Species  \\\n",
       "0          NaN              NaN         37.1   1  ...  diploid  Homo sapiens   \n",
       "1          NaN              NaN         37.1   1  ...  diploid  Homo sapiens   \n",
       "2          NaN              NaN         37.1   1  ...  diploid  Homo sapiens   \n",
       "3          NaN              NaN         37.1  10  ...  diploid  Homo sapiens   \n",
       "4          NaN              NaN         37.1  10  ...  diploid  Homo sapiens   \n",
       "\n",
       "        Source SourceVersion  SourceStrand  \\\n",
       "0  1000genomes           0.0          PLUS   \n",
       "1  1000genomes           0.0          PLUS   \n",
       "2  1000genomes           0.0          PLUS   \n",
       "3  1000genomes           0.0          PLUS   \n",
       "4  1000genomes           0.0          PLUS   \n",
       "\n",
       "                                           SourceSeq  \\\n",
       "0  GAGGGCCGCTCAGCGAGGGCGGGACAGAACCTCTCCCGGGCTGGGA...   \n",
       "1  CATGGAGCTGTGTTTGTACCCCACCTCCAAATTCCACCACTGGCCC...   \n",
       "2  CTTTATGCAGCCCAAGTCAGAACCATGGCTCCAAAACAAAAGAAAA...   \n",
       "3  GGGAACACCTGGCTCTGCAAGTTTTGTGCCTTAGCGGAAAGAGGCG...   \n",
       "4  CTGTGTTTATTTGGTTCCTTAGGTAAAGCCATTAGCTTGTTAGCAA...   \n",
       "\n",
       "                                       TopGenomicSeq BeadSetID  Exp_Clusters  \\\n",
       "0  GAGGGCCGCTCAGCGAGGGCGGGACAGAACCTCTCCCGGGCTGGGA...     662.0           3.0   \n",
       "1  CATGGAGCTGTGTTTGTACCCCACCTCCAAATTCCACCACTGGCCC...     662.0           3.0   \n",
       "2  CTTTATGCAGCCCAAGTCAGAACCATGGCTCCAAAACAAAAGAAAA...     688.0           3.0   \n",
       "3  GGGAACACCTGGCTCTGCAAGTTTTGTGCCTTAGCGGAAAGAGGCG...     662.0           3.0   \n",
       "4  CTGTGTTTATTTGGTTCCTTAGGTAAAGCCATTAGCTTGTTAGCAA...     662.0           3.0   \n",
       "\n",
       "   RefStrand  \n",
       "0          -  \n",
       "1          -  \n",
       "2          +  \n",
       "3          +  \n",
       "4          -  \n",
       "\n",
       "[5 rows x 21 columns]"
      ]
     },
     "execution_count": 140,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "GPL_prime.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 141,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "La première valeur de IlmnID est exm-IND1-200449980-0_M_R_1990486447\n",
      "La première valeur de Name est exm-IND1-200449980\n",
      "La première valeur de IlmnStrand est MINUS\n",
      "La première valeur de SNP est D/I\n",
      "La première valeur de AddressA_ID est 69606251.0\n",
      "La première valeur de AlleleA_ProbeSeq est CTGTGGAACTTTCCGGAGCTCGGCACCCTTCCCCCAAGCCTACCTGGGCG\n",
      "La première valeur de AddressB_ID est nan\n",
      "La première valeur de AlleleB_ProbeSeq est nan\n",
      "La première valeur de GenomeBuild est 37.1\n",
      "La première valeur de Chr est 1\n",
      "La première valeur de MapInfo est 202183358.0\n",
      "La première valeur de Ploidy est diploid\n",
      "La première valeur de Species est Homo sapiens\n",
      "La première valeur de Source est 1000genomes\n",
      "La première valeur de SourceVersion est 0.0\n",
      "La première valeur de SourceStrand est PLUS\n",
      "La première valeur de SourceSeq est GAGGGCCGCTCAGCGAGGGCGGGACAGAACCTCTCCCGGGCTGGGAGTGCACGGCGCGGT[-/G]CGCCCAGGTAGGCTTGGGGGAAGGGTGCCGAGCTCCGGAAAGTTCCACAGGTCCTCCGCA\n",
      "La première valeur de TopGenomicSeq est GAGGGCCGCTCAGCGAGGGCGGGACAGAACCTCTCCCGGGCTGGGAGTGCACGGCGCGGT[-/G]CGCCCAGGTAGGCTTGGGGGAAGGGTGCCGAGCTCCGGAAAGTTCCACAGGTCCTCCGCA\n",
      "La première valeur de BeadSetID est 662.0\n",
      "La première valeur de Exp_Clusters est 3.0\n",
      "La première valeur de RefStrand est -\n"
     ]
    }
   ],
   "source": [
    "#Convertir le dataframe GPL_prime en dictionnaire ou chaque clé représente le nom de la colonne et les valeurs représente la valeur de chaque colonne\n",
    "dict_GPL=GPL_prime.to_dict(orient='list')\n",
    "for key in dict_GPL:\n",
    "    print(f\"La première valeur de {key} est {dict_GPL[key][0]}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 142,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Convertir le dtaframe GSE_prime en dictionnaire\n",
    "dict_GSE=GSE_prime.to_dict(orient='list')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "metadata": {},
   "outputs": [],
   "source": [
    "def keep_keys_only(dictionnaire, liste_cles):\n",
    "    \"\"\"\n",
    "    Cette fonction garde une liste de clés dans un dictionnaire et supprime toutes les autres clés et leurs valeurs.\n",
    "\n",
    "    Args:\n",
    "        dictionnaire (dict): le dictionnaire à modifier.\n",
    "        liste_cles (list): une liste de clés à garder.\n",
    "\n",
    "    Returns:\n",
    "        dict: le dictionnaire modifié.\n",
    "    \"\"\"\n",
    "    for cle in list(dictionnaire.keys()):\n",
    "        if cle not in liste_cles:\n",
    "            del dictionnaire[cle]\n",
    "    return dictionnaire\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 143,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "La première valeur de IlmnID est exm-IND1-200449980-0_M_R_1990486447\n",
      "La première valeur de SNP est D/I\n",
      "La première valeur de Chr est 1\n",
      "La première valeur de MapInfo est 202183358.0\n",
      "La première valeur de RefStrand est -\n"
     ]
    }
   ],
   "source": [
    "#Redéfinir dict_GPL en gardant que les colonnes qui nous intéresse\n",
    "cle_a_garder = [\"IlmnID\", \"SNP\", \"Chr\", \"MapInfo\", \"RefStrand\"]\n",
    "dictionnaire = keep_keys_only(dict_GPL, cle_a_garder)\n",
    "#Tester la fonction keep_keys_only\n",
    "for key in dictionnaire:\n",
    "    print(f\"La première valeur de {key} est {dictionnaire[key][0]}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {},
   "outputs": [
    {
     "ename": "SyntaxError",
     "evalue": "invalid syntax (<ipython-input-104-0d15b8370d96>, line 1)",
     "output_type": "error",
     "traceback": [
      "\u001b[0;36m  File \u001b[0;32m\"<ipython-input-104-0d15b8370d96>\"\u001b[0;36m, line \u001b[0;32m1\u001b[0m\n\u001b[0;31m    tqdm(for i in (range(0, len(GPL.index))):\u001b[0m\n\u001b[0m           ^\u001b[0m\n\u001b[0;31mSyntaxError\u001b[0m\u001b[0;31m:\u001b[0m invalid syntax\n"
     ]
    }
   ],
   "source": [
    "for i in tqdm(range(0, len(GPL.index))):\n",
    "    # Conversion des génotypes en format VCF\n",
    "    hapmap.loc[i,\"exm#\"] = GPL.loc[i,\"IlmnID\"]\n",
    "    hapmap.loc[i, \"alleles\"] = GPL.loc[i,\"SNP\"]\n",
    "    hapmap.loc[i,\"chrom\"] = GPL.loc[i,\"Chr\"]\n",
    "    hapmap.loc[i,\"pos\"] = GPL.loc[i,\"MapInfo\"]\n",
    "    hapmap.loc[i,\"strand\"] = GPL.loc[i,\"RefStrand\"]\n",
    "    hapmap.loc[i,\"assembly#\"] = \"NA\"\n",
    "    hapmap.loc[i,\"center\"] = \"NA\"\n",
    "    hapmap.loc[i,\"protLSID\"] = \"NA\"\n",
    "    hapmap.loc[i,\"assayLSID\"] = \"NA\"\n",
    "    hapmap.loc[i,\"panelLSID\"] = \"NA\"\n",
    "    hapmap.loc[i,\"QCcode\"] = \"NA\"\n",
    "    \"\"\"for j in range (11,len(hapmap.columns)):\n",
    "        line=extract_row_by_id(GSE, GPL.loc[i,\"IlmnID\"], \"ID_REF\")\n",
    "        hapmap.iloc[i,j] = extract_cell_value(line,j-11)\"\"\"\n",
    "\n",
    "# Écriture des génotypes convertis en format hapmap dans un fichier\n",
    "#hapmap.to_csv(\"data.hapmap\", sep=\"\\t\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
